smithy-linters/src/main/java/software/amazon/smithy/linters/CamelCaseValidator.java

```diff
@@ -135,14 +157,66 @@ public List<ValidationEvent> validate(Model model) {
                         shape.getType(), shape.getId().getName())))
                 .forEach(events::add);
 
-        Pattern isValidMemberName = config.getMemberNames().getRegex();
-        model.shapes(MemberShape.class)
-                .filter(shape -> !isValidMemberName.matcher(shape.getMemberName()).find())
-                .map(shape -> danger(shape, format(
-                        "Member shape member name, `%s`, is not %s camel case",
-                        shape.getMemberName(), config.getMemberNames())))
-                .forEach(events::add);
+        // First validate all member shapes connected to services
+        Set<Shape> serviceClosure = new HashSet<>();
+        for (ServiceShape serviceShape : model.getServiceShapes()) {
+            Walker walker = new Walker(model);
+            walker.iterateShapes(serviceShape).forEachRemaining(serviceClosure::add);
```

Comment:
You should get the set of shapes in the closure and validate those shapes. Then also add all the shapes found in the closure to `serviceClosure` to exclude it from the namespace checks.